knpu/episode3/broken-ship.md

```diff
@@ -9,20 +9,31 @@ you just need to extend `AbstractShip`.
 
 What's really great is that `AbstractShip` now tells you exactly what you need
 to do to create a new ship class with its three abstract functions that you must
-fill in.
+fill in:
+
+[[[ code('4d7153990d') ]]]
 
 A third group has joined the battle and we have a new type of ship. They're not
 very good mechanics, so we'll call this a broken ship. This is simple, the ship
 is always broken. 
 
-Create a new php class called `BrokenShip`. Of course now make it extend `AbstractShip`.
+Create a new php class called `BrokenShip`. Of course now make it extend `AbstractShip`:
+
+[[[ code('19f350c07d') ]]]
+
 Let's pretend like we don't know that there are any abstract methods in the parent class.
-So we won't do anything here except putting in the extends code. Head over to `Bootstrap.php`
-and require our useless new `BrokenShip`. 
+So we won't do anything here except putting in the extends code. Head over to `bootstrap.php`
+and require our useless new `BrokenShip`:
+
+[[[ code('ac130fc069') ]]]
  
 Back in `index.php` for now, let's just add `$brokenShip = new BrokenShip();` and add it 
-to our ships array. We can do this because we know that `BrokenShip` extends `AbstractShip`. 
-And down here, when we use those ship objects we're just calling methods on the abstract ship.
+to our ships array:
+
+[[[ code('be307a1387') ]]]
```

Comment:
Good job hearing highlighting `$brokenShip = new BrokenShip('Just a hunk of metal');`. Sometimes I make a mistake in the screencast recording, but it's so silly, that I don't go back and update the code with the same mistake. In those cases - exactly like you did - I just show the *right* way in the code block.